features/extensions/azure/install.py

```diff
@@ -83,30 +83,46 @@ def install(jazz_stackprefix, jenkins_url, jenkins_user, jenkins_api_token, jazz
     metricJobUrl = "job/build-pack-api/buildWithParameters?token=jazz-101-job&service_name=metrics&domain" \
                    "=jazz&scm_branch=master"
     startJob(jenkins_url, jenkins_user, jenkins_api_token, metricJobUrl)
     # Trigger jazz ui
-    startJob(jenkins_url, jenkins_user, jenkins_api_token, "job/jazz_ui/build?token=jazz-101-job")
+    startJob(jenkins_url, jenkins_user, jenkins_api_token, "job/jazz_ui/buildWithParameters?token=jazz-101-job")
 
 
-def prepare_azure_json(azure_location):
+def prepare_azure_json(azure_location, azure_subscription_id):
     extension_base = "extensions/azure"
     azureConfig = OrderedDict()
     azureConfig['IS_ENABLED'] = True
-    azureConfig['SUBSCRIPTION_ID'] = 'AZ_SUBSCRIPTIONID'
-    azureConfig['CLIENT_ID'] = 'AZ_CLIENTID'
-    azureConfig["PASSWORD"] = 'AZ_PASSWORD'
-    azureConfig["TENANT_ID"] = 'AZ_TENANTID'
-    azureConfig['LOCATION'] = azure_location
-    azureConfig['RESOURCE_GROUPS'] = OrderedDict()
-    azureConfig['RESOURCE_GROUPS']['DEVELOPMENT'] = getTerraformOutputVar("dev_resource_group", extension_base)
-    azureConfig['RESOURCE_GROUPS']['STAGING'] = getTerraformOutputVar("stage_resource_group", extension_base)
-    azureConfig['RESOURCE_GROUPS']['PRODUCTION'] = getTerraformOutputVar("prod_resource_group", extension_base)
-    azureConfig['APIM'] = OrderedDict()
-    azureConfig['APIM']['DEVELOPMENT'] = getTerraformOutputVar("dev_apim", extension_base)
-    azureConfig['APIM']['STAGING'] = getTerraformOutputVar("stage_apim", extension_base)
-    azureConfig['APIM']['PRODUCTION'] = getTerraformOutputVar("prod_apim", extension_base)
-
+    azureAccount = OrderedDict()
+    azureAccount['ACCOUNTID'] = azure_subscription_id
+    azureAccount['ACCOUNTNAME'] = 'azureaccount'
+    azureAccount['SUBSCRIPTION_ID'] = 'AZ_SUBSCRIPTIONID'
+    azureAccount['CLIENT_ID'] = 'AZ_CLIENTID'
+    azureAccount["PASSWORD"] = 'AZ_PASSWORD'
+    azureAccount["TENANT_ID"] = 'AZ_TENANTID'
+    azureRegions = OrderedDict()
+    azureRegions['LOCATION'] = azure_location
+    azureRegions['REGION'] = azure_location.lower().replace(' ','')
+    azureRegions['RESOURCE_GROUPS'] = OrderedDict()
+    azureRegions['RESOURCE_GROUPS']['DEV'] = getTerraformOutputVar("dev_resource_group", extension_base)
+    azureRegions['RESOURCE_GROUPS']['PROD'] = getTerraformOutputVar("prod_resource_group", extension_base)
+    azureRegions['RESOURCE_GROUPS']['STG'] = getTerraformOutputVar("stage_resource_group", extension_base)
+    azureRegions['APIM'] = OrderedDict()
+    azureRegions['APIM']['DEV'] = getTerraformOutputVar("dev_apim", extension_base)
+    azureRegions['APIM']['PROD'] = getTerraformOutputVar("prod_apim", extension_base)
+    azureRegions['APIM']['STG'] = getTerraformOutputVar("stage_apim", extension_base)
+    azureAccount['REGIONS'] = [azureRegions]
+    azureConfig['ACCOUNTS'] = [azureAccount]
+    azureDefault = OrderedDict()
+    azureDefault['ACCOUNTID'] = azure_subscription_id
```

Comment:
Note - While this (assigning these values to `azureDefault` always) may work now, we need to revisit this piece of code when we add the ability to support multiple azure subscriptions & regions.

Reply:
Right. We will have to make some major changes on Azure Installer and Azure build packs to support multiple subscriptions/regions. We will address this then. There is no benefit in trying to remove it now. We just kept it there to be consistent with AWS.